src/api/status/public/js/build-log/check-for-build.js

```diff
@@ -1,48 +1,47 @@
-/* eslint-disable consistent-return */
-import { checkBuildStatus, getBuildLog } from './api.js';
+import checkBuildStatus from './check-build-status.js';
 import terminal from './terminal.js';
 import buildHeader from './build-header.js';
 
-let build;
-let reader;
+export default async function checkForBuild() {
+  const status = await checkBuildStatus();
 
-async function finish() {
-  try {
-    await reader.cancel();
-  } finally {
-    build = null;
-  }
-}
+  // Prefer the current build, but fallback to the previous one
+  const build = status.current ?? status.previous;
 
-function processLog({ done, value }) {
-  if (done) {
-    return finish();
-  }
+  // Render the build header info
+  buildHeader(build);
 
-  if (terminal) {
-    terminal.write(value);
+  if (!build) {
+    return;
   }
 
-  // Recursively invoke processLog until `done === true`
-  return reader.read().then(processLog).catch(finish);
-}
-
-export default async function checkForBuild() {
-  const status = await checkBuildStatus();
-  buildHeader(status);
-
-  // If we're already building, skip this check
-  if (build) {
+  const reader = await build.getReader();
+  if (!reader) {
     return;
   }
 
-  if (status.building) {
-    terminal.clear();
-    reader = await getBuildLog();
-    if (reader) {
-      // eslint-disable-next-line require-atomic-updates
-      build = { reader, title: status.title, startedAt: status.startedAt };
-      reader.read().then(processLog).catch(finish);
+  const finish = async () => {
+    try {
+      await reader.cancel();
+    } catch (err) {
+      console.warn('Unable to clean up build log reader');
     }
-  }
+  };
+
+  const processLog = ({ done, value }) => {
+    if (done) {
+      return finish();
+    }
+
+    // Write the data to the terminal
+    terminal.write(value);
+
+    // Recursively invoke processLog until `done === true`
+    return reader.read().then(processLog).catch(finish);
+  };
+
+  terminal.clear();
+
+  // Start processing the log output messages
+  reader.read().then(processLog).catch(finish);
```

Comment:
Would this be a bit cleaner. 
```suggestion js
const processLog = () => {
    reader
      .read()
      .then(({ done, value }) => {
        if (done) {
          return finish();
        }
        // Write the data to the terminal
        terminal.write(value);
        // Recursively invoke processLog until `done === true`
        return processLog();
      })
      .catch(finish);
  };

  // Start processing the log output messages
  processLog();
```